### PR TITLE
Updated Github actions to test on previous major versions

### PR DESCRIPTION
**Pull request by LeviSklaroff**
**Issue #:**
Github actions are required to be updated to test old release builds whenever we create a new major version
**Description of changes:**
Updated actions to automatically test on old versions
**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.* No

**Checklist:**

1. Have you successfully run `npm run build:release` locally? Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ltrung** [comment]
>Updated github workflows to test on old major versions

nit: can we change it to previous major version? 2.x is not really old :D

File: .github/workflows/browser-compatibility-test-previous-major-version.yml
Comment (ltrung) [inline .github/workflows/browser-compatibility-test-previous-major-version.yml:33]: Does this syntax work? I tried `echo "$((3.0.0-1))"` and it threw an arithmetic error for me.

Comment (LeviSklaroff) [reply .github/workflows/browser-compatibility-test-previous-major-version.yml:33]: Ah I'll have to edit it then, I thought it just outputted the first number

**Comment by ltrung** [comment]
Will this run against release-1.x once merged? Note that a lot of tests are not available in 1.x branch so you may want to disable it (background blur, vf, checker, etc.)

**Comment by LeviSklaroff** [comment]
> Will this run against release-1.x once merged? Note that a lot of tests are not available in 1.x branch so you may want to disable it for now (background blur, vf, checker, etc.) or we can wait till we have v3 branch.

This is supposed to run 1.x at the moment, probably best to just not merge it till v3 and I'll just expect them to fail while testing the workflow.

**Comment by ltrung** [comment]
> > Will this run against release-1.x once merged? Note that a lot of tests are not available in 1.x branch so you may want to disable it for now (background blur, vf, checker, etc.) or we can wait till we have v3 branch.
> 
> This is supposed to run 1.x at the moment, probably best to just not merge it till v3 and I'll just expect them to fail while testing the workflow.

If so can we split this into 2 PR. The other GitHub actions for CI can be merged now :)

**Comment by LeviSklaroff** [comment]
> > > Will this run against release-1.x once merged? Note that a lot of tests are not available in 1.x branch so you may want to disable it for now (background blur, vf, checker, etc.) or we can wait till we have v3 branch.
> > 
> > 
> > This is supposed to run 1.x at the moment, probably best to just not merge it till v3 and I'll just expect them to fail while testing the workflow.
> 
> If so can we split this into 2 PR. The other GitHub actions for CI can be merged now :)

That's a good idea! I'll do that.

**Comment by nainkunal933** [comment]
Why are you not testing previous major versions for `audio-video-transform-integration.yml` https://github.com/aws/amazon-chime-sdk-js/blob/main/.github/workflows/audio-video-transform-integration.yml

File: .github/workflows/audio-integration.yml
Comment (nainkunal933) [inline .github/workflows/audio-integration.yml:7]: This would test against any of the older major releases for e.g. if that is 1.x, 2.x or 3.x. How would I exclude a previous major release in case of a very old version that is deprecated?

**Review by nainkunal933 — CHANGES_REQUESTED** [review]
Rather than ** which is a wild character can you please use [] to match the previous major versions. [] will give us more flexibility. Ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

For example, release-[1-2].x

File: .github/workflows/audio-integration.yml
Comment (ltrung) [reply .github/workflows/audio-integration.yml:7]: We can delete the branch basically. If we deprecate a version there is no reason to keep the branch.

**Review by ltrung — CHANGES_REQUESTED** [review]
What about `audio-video-transform-integration.yml` ?

**Comment by ltrung** [comment]
> Rather than ** which is a wild character can you please use [] to match the previous major versions. [] will give us more flexibility. Ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
> 
> For example, release-[1-2].x

Prefer ** for now. We can always change it later.

**Comment by nainkunal933** [comment]
> > Rather than ** which is a wild character can you please use [] to match the previous major versions. [] will give us more flexibility. Ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
> > For example, release-[1-2].x
> 
> Prefer ** for now. We can always change it later.

Any reasons why you prefer ** over []? I like [] because I can exclude a branch without having to delete it.

**Comment by ltrung** [comment]
> > > Rather than ** which is a wild character can you please use [] to match the previous major versions. [] will give us more flexibility. Ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
> > > For example, release-[1-2].x
> > 
> > 
> > Prefer ** for now. We can always change it later.
> 
> Any reasons why you prefer ** over []? I like [] because I can exclude a branch without having to delete it.

Because we don't need those flexibility. We will delete release-1.x next week.

**Comment by LeviSklaroff** [comment]
> Why are you not testing previous major versions for `audio-video-transform-integration.yml` https://github.com/aws/amazon-chime-sdk-js/blob/main/.github/workflows/audio-video-transform-integration.yml

I'll add this, I just missed it.

File: .github/workflows/audio-integration.yml
Comment (LeviSklaroff) [reply .github/workflows/audio-integration.yml:7]: There's also syntax to exclude branches if we want to, if we put in !release-1.x that will exclude that branch from being ran on.

**Review by ltrung — APPROVED** [review]

**Pull request merged**
